data_model/src/domain.rs

```diff
@@ -62,9 +69,12 @@ impl From<GenesisDomain> for Domain {
 /// Builder which can be submitted in a transaction to create a new [`Domain`]
 #[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
 pub struct NewDomain {
+    /// The identification associated to the domain builder.
     id: <Domain as Identifiable>::Id,
+    /// The (IPFS) link to the logo of this domain.
     logo: Option<IpfsPath>,
-    metadata: Metadata,
+    /// metadata associated to the domain builder.
+    pub metadata: Metadata,
```

Comment:
we cannot have public fields. There is a `with_metadata` that you can use to add metadata to your domain builder

Reply:
I can imagine you need to inspect the contents of `NewDomain` for the generator tool. You can do that by building `Domain` first with `NewDomain::build` and then you will have `Domain::get_metadata` available